ui/app/hooks/useTokenTracker.js

```diff
@@ -1,8 +1,17 @@
 import { useState, useEffect, useRef, useCallback } from 'react'
 import TokenTracker from '@metamask/eth-token-tracker'
 import { useSelector } from 'react-redux'
+import { isEqual } from 'lodash'
 import { getCurrentNetwork, getSelectedAddress } from '../selectors'
 
+const usePrevious = (value) => {
+  const ref = useRef()
+  useEffect(() => {
```

Comment:
I could be horribly wrong here, but I believe the order of effects matters. These effects will happen before the effect that is checking their value. For this, I think that means they will *always* be equal because the current value is set before you're checking it on line 80. The result would be fixing the infinite renders bug but potentially breaking the effect of reloading tokens when the values change? 

I think what we want is to always return a consistent value for use in the dependency array of the effect in question, similar to passing isEqual to the second argument of useSelector. 

https://github.com/reduxjs/react-redux/blob/a9dce030742385285a243b8769f49c94c5942838/src/hooks/useSelector.js#L30-L74

```js
function useMemoizedValue(value, equalityFn = isEqual) {
  const [, forceRender] = useReducer(s => s + 1, 0)
  const latestValue = useRef();

  useIsomorphicLayoutEffect(() => {
	if (!isEqual(value, latestValue.current)) {
      latestValue.current = value;
      forceRender(); // refs don't cause renders
    }
  }, [value]);

  return latestValue.current;
}
```

downstream usage:

```js
const memoizedTokens = useMemoizedValue(tokens);
```

Disclaimer: I did not test this, but I feel like there's some significant potential here as a tool to prevent unnecessary renders that can potentially impact user experience. We use the second argument of useSelector in other parts of the code base with great effect. CC'ing @Gudahtt here to get a second opinion.